packages/worker/src/worker.ts

```diff
@@ -229,17 +229,26 @@ export class Worker {
     const constructedWorker = await this.bundleWorker(compiledOptions, replayWorker);
 
     const runPromise = constructedWorker.run();
-    const pollerShutdown = constructedWorker.workflowPollerStateSubject.pipe(
-      filter((state): state is 'SHUTDOWN' => state === 'SHUTDOWN')
-    );
+    const pollerShutdown = constructedWorker.workflowPollerStateSubject.pipe(filter((state) => state !== 'POLLING'));
     pollerShutdown.subscribe((_) => {
```

Comment:
Let's use `firstValueFrom` instead of subscribing.
If you subscribe you need to unsubscribe too which is more error prone (e.g. I missed it in the first 2 reviews).
https://rxjs.dev/api/index/class/Observable#cancel-a-subscription